### ISSUE TITLE
Missing files in the WordPress repository?...

### ISSUE THREAD

**Issue by GwynethLlewelyn** [dartiss/plugin-readme-parser#32]
I thought I was going insane when I got the following error in my logs:

```
PHP Warning:  include_once(): Failed opening '[...]/wp-content/plugins/wp-readme-parser/includes/admin-config.php' for inclusion (include_path='.:[...]/') in [...]//wp-content/plugins/wp-readme-parser/wp-readme-parser.php
```
(paths redacted for privacy)

And, indeed, that file was missing. Weird. I checked here on GitHub, and, sure thing, that file _is_ here (and probably always was!).

So I've downloaded it and looked at the errors log again... and now another one was missing:

```
PHP Warning:  include_once(): Failed opening '[...]/wp-content/plugins/wp-readme-parser/includes/admin-config.php' for inclusion (include_path='.:[...]/') in [...]/wp-content/plugins/wp-readme-parser/wp-readme-parser.php on line 36
```
What?! But, again, this seemed to be correct: that file was missing, too!

This time I thought that there was something amiss with the plugin, so I deactivated it, removed it, made sure the directory was gone, then installed it fresh from the WordPress Plugin Library, activated it and... the errors appeared again.. and — surprise, surprise! — _both_ files were missing!

So, the problem is not on GitHub — where everything is fine — but somehow the upload to the Plugin Library must have deleted those files somehow, during the last update...

Oh, and while you're working on it... you may bump the PHP Markdown library from @Michelf to its latest incarnation, namely, 2.0.0 (from September 2022). There seem to be no other updates since then.

While #22 mentions the upgrade to 1.9, the version inside Markdown.php still says 1.6.0 — I haven't checked if the rest of the code is 1.9 or 1.6.0, but, in any case, 2.0.0 has been around for over a year now, and it's reasonably safe to assume that whatever bugs it might have had in the past have all been successfully eliminated 😁 The only issue might be dealing with potential disruption in bumping up a _major_ version (who knows if the method calls have changed since the 1.X series...?).

Also mmh, I hate to recommend something that I use so seldom, but when dealing with dependencies of third-party vendors, it would be better to:

- either use [Composer](https://getcomposer.org/) calling [PHP Markdown on Packagist](https://packagist.org/packages/michelf/php-markdown) to keep those dependencies in sync;
- _or_ if you cannot bother with PHP package management (I feel your pain!), just include Michelf's GitHub repository as a [submodule](https://git-scm.com/book/en/v2/Git-Tools-Submodules) — and configure everything at the Git level to deal automagically with fetching the required files from Michelf's repository.

(I know — I won't insist in either of those two methods, since I'm usually also way too lazy to use them on my own projects and WP plugins... so the above advice also applies to my ancient things, some of which _do_ require third-party libraries.)

**Comment by morehawes** [comment]
Hi @GwynethLlewelyn,

Thank you so much for your input and recommendations! I really appreciate it. Also my apologies in the delay in getting back to you - it seems that somehow I don't have notifications set on this repository yet since adopting the plugin last week :0)

> So, the problem is not on GitHub — where everything is fine — but somehow the upload to the Plugin Library must have deleted those files somehow, during the last update...

Yes I think you are correct. During the transfer last week I ran into some SVN issues and had to get a plugin moderator involved in order to remedy it. It's definitely possible that that was the cause.

I will try to get this resolved ASAP.

> Oh, and while you're working on it... you may bump the PHP Markdown library from @michelf to its latest incarnation, namely, 2.0.0 (from September 2022). There seem to be no other updates since then.

While I use the plugin on my own site, I am not yet familiar with the codebase; so I plan on sitting down in the near future and getting my head around what is there. PHP 8 compatibility will be the main driver and I have seen issues and the [open PR](https://github.com/morehawes/plugin-readme-parser/pull/25) relating to the Markdown library.

> (I know — I won't insist in either of those two methods, since I'm usually also way too lazy to use them on my own projects and WP plugins... so the above advice also applies to my ancient things, some of which do require third-party libraries.)

I'm not apposed to using Composer, but definitely lean towards the sub-module route!

Thanks again for your input and patience during this transfer period :) PRs are definitely welcome!

**Comment by GwynethLlewelyn** [comment]
Awesome! Thanks for looking into it :) I mean, it's not as if the Earth suddenly stops spinning — or that I lose a few subscribers to my blog, where I use your plugin (it has _none_. Well, many bots are regulars :) ). But whenever you've got time....

I'd love to do a PR if I had time to work on yet _another_ project (no matter how small it is... they tend to pile up and creep on you....), and, in any case, it _should_ be easy enough to deal with. I think. I haven't tried :)

**Comment by morehawes** [comment]
Fixed in v1.3.11. Please keep an eye on the plugin for future updates and feedback welcome :)

Joe
